components/raftstore/src/store/snap.rs

```diff
@@ -493,36 +439,22 @@ impl Snap {
         Ok(s)
     }
 
-    pub fn new_for_building<T: Into<PathBuf>>(
+    fn new_for_building<T: Into<PathBuf>>(
         dir: T,
         key: &SnapKey,
-        size_track: Arc<AtomicU64>,
-        deleter: Box<dyn SnapshotDeleter>,
-        limiter: Limiter,
-        mgr: Option<Arc<DataKeyManager>>,
+        mgr: Arc<RwLock<SnapManagerCore>>,
     ) -> RaftStoreResult<Self> {
-        let mut s = Self::new(dir, key, size_track, true, true, deleter, limiter, mgr)?;
+        let mut s = Self::new(dir, key, true, true, mgr)?;
         s.init_for_building()?;
```

Comment:
There is no lock to protect building anymore, will it become a problem?

Reply:
Why? the lock scope is hot changed for now.